src/features/learning/components/PracticeCards/PracticeSelectImage/PracticeSelectImage.tsx

```diff
@@ -0,0 +1,74 @@
+import {typedMemo} from "../../../../../core/utils/typedMemo";
+import React, {FC, useEffect, useState} from "react";
+import styles from "./PracticeSelectImage.module.css";
+import clsx from "clsx";
+import {ComponentProps} from "../../../../../core/models/ComponentProps";
+import PracticeIconSVG from "../../../../../assets/images/PracticeIcon.svg"
+import {LearningBlock} from "../../LearningBlock";
+import {Word} from "../../../../../core/models/Word";
+import {SignVideo} from "../../../../../components/SignVideo";
+import {Typography} from "../../../../../components/Typography";
+import {SelectImage} from "../../SelectEntity/SelectImage";
+import {StepStatus} from "../../../../../core/models/StepStatus";
+import {shuffleArray} from "../../../../../core/utils/shuffleArray";
+import {getSelectEntity} from "../../SelectEntity/SelectContainer/SelectContainer";
+
+type Props = ComponentProps & Readonly<{
+    wordObject: Word;
+    otherVariants: Word[];
+    checked: boolean;
+    setStatus: React.Dispatch<React.SetStateAction<StepStatus>>;
+    setIsTaskReadyToCheck: React.Dispatch<React.SetStateAction<boolean>>;
+}>
+
+/** Практика "Выбери изображение". */
+export const PracticeSelectImage: FC<Props> = typedMemo(function PracticeSelectImage(props) {
+    const [selectImage, setSelectImage] = useState<Word | null>()
+    const [variants] = useState(shuffleArray([props.wordObject, ...props.otherVariants]))
+
+    useEffect(() => {
+        if (props.checked) {
```

Comment:
Чтобы убрать вложенность условий, можно в начале прописать такое:
```suggestion
        if (!props.checked) {
           return
        }
```